src/main/java/com/mentalhealthapi/mentalhealthapi/MentalHealthController.java

```diff
@@ -29,15 +29,19 @@ public String index() {
      */
     @GetMapping("/disorders")
     public ResponseEntity fetchAllDisorders() {
-        return new ResponseEntity(HttpStatus.OK);
+        try {
+            return new ResponseEntity(HttpStatus.OK);
+        } catch (Exception e) {
+            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
     }
 
     /**
      * Fetch a disorder by name
      * @return specific disorder page
      */
     @GetMapping("/disorders/{name}/")
-    public ResponseEntity fetchDisorder(@PathVariable("name") String name) {
+    public ResponseEntity fetchDisorderByName(@PathVariable("name") String name) {
```

Comment:
I found that you could probably be slightly more clear on the naming, here. This name provides more information on what it is that needs to be passed to the endpoint.

Reply:
Actually... the parameter name is descriptive enough.   It's fine either way.

Reply:
fetchDisorderByName actually is more readable to me and makes sense instantly. I great suggestion